Running any gh cli command on my local machine causing a seg fault/goroutine error

Running any gh cli command causes a seg fault. I have uninstalled and reinstalled the gh cli, removed all node.js/ npm from my machine, and neither of these things has solved the issue.

```
(base) yamini@Yaminis-MacBook-Pro local % gh auth login
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1031c6580]

goroutine 1 [running]:
github.com/cli/go-gh/pkg/config.(*Config).Get(0x1038a3548?, {0x140007bfbd8?, 0x30?, 0x10488aea0?})
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/config/config.go:46 +0x30
github.com/cli/go-gh/pkg/browser.resolveLauncher()
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:75 +0x6c
github.com/cli/go-gh/pkg/browser.New(...)
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:32
github.com/cli/cli/v2/internal/browser.New({0x0, 0x0}, {0x1049098a0, 0x14000010018}, {0x1049098a0, 0x14000010020})
    github.com/cli/cli/v2/internal/browser/browser.go:14 +0x58
github.com/cli/cli/v2/pkg/cmd/factory.newBrowser(0x140003bfdc0?)
    github.com/cli/cli/v2/pkg/cmd/factory/default.go:111 +0x50
github.com/cli/cli/v2/pkg/cmd/factory.New({0x10477ac48, 0x6})
    github.com/cli/cli/v2/pkg/cmd/factory/default.go:37 +0x238
main.mainRun()
    github.com/cli/cli/v2/cmd/gh/main.go:64 +0xec
main.main()
    github.com/cli/cli/v2/cmd/gh/main.go:48 +0x1c
``` 
I get the same error with any cli command, not just auth. Another example, gh repo create:

```
(base) yamini@Yaminis-MacBook-Pro leet % gh repo create
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100622580]

goroutine 1 [running]:
github.com/cli/go-gh/pkg/config.(*Config).Get(0x100cff548?, {0x140007dfbd8?, 0x30?, 0x101ce6ea0?})
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/config/config.go:46 +0x30
github.com/cli/go-gh/pkg/browser.resolveLauncher()
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:75 +0x6c
github.com/cli/go-gh/pkg/browser.New(...)
    github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:32
github.com/cli/cli/v2/internal/browser.New({0x0, 0x0}, {0x101d658a0, 0x140001aa008}, {0x101d658a0, 0x140001aa010})
    github.com/cli/cli/v2/internal/browser/browser.go:14 +0x58
github.com/cli/cli/v2/pkg/cmd/factory.newBrowser(0x140000f7ce0?)
    github.com/cli/cli/v2/pkg/cmd/factory/default.go:111 +0x50
github.com/cli/cli/v2/pkg/cmd/factory.New({0x101bd6c48, 0x6})
    github.com/cli/cli/v2/pkg/cmd/factory/default.go:37 +0x238
main.mainRun()
    github.com/cli/cli/v2/cmd/gh/main.go:64 +0xec
main.main()
    github.com/cli/cli/v2/cmd/gh/main.go:48 +0x1c
```

I can't check gh --version because that causes a seg fault too:

```
(base) yamini@Yaminis-MacBook-Pro leet % gh --version  
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102b46580]

goroutine 1 [running]:
github.com/cli/go-gh/pkg/config.(*Config).Get(0x103223548?, {0x140008dfbd8?, 0x30?, 0x10420aea0?})
	github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/config/config.go:46 +0x30
github.com/cli/go-gh/pkg/browser.resolveLauncher()
	github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:75 +0x6c
github.com/cli/go-gh/pkg/browser.New(...)
	github.com/cli/go-gh@v0.1.1-0.20220817122932-3630ab390fe7/pkg/browser/browser.go:32
github.com/cli/cli/v2/internal/browser.New({0x0, 0x0}, {0x1042898a0, 0x140001aa008}, {0x1042898a0, 0x140001aa010})
	github.com/cli/cli/v2/internal/browser/browser.go:14 +0x58
github.com/cli/cli/v2/pkg/cmd/factory.newBrowser(0x14000163ce0?)
	github.com/cli/cli/v2/pkg/cmd/factory/default.go:111 +0x50
github.com/cli/cli/v2/pkg/cmd/factory.New({0x1040fac48, 0x6})
	github.com/cli/cli/v2/pkg/cmd/factory/default.go:37 +0x238
main.mainRun()
	github.com/cli/cli/v2/cmd/gh/main.go:64 +0xec
main.main()
	github.com/cli/cli/v2/cmd/gh/main.go:48 +0x1c
```
I haven't made any changes to the .git/config file and afaik, I don't have any unusual configurations.

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -24,8 +24,9 @@ const (
 )
 
 var (
-	cfg  *Config
-	once sync.Once
+	cfg     *Config
+	once    sync.Once
+	loadErr error
 )
 
 // Config is a in memory representation of the gh configuration files.
@@ -121,11 +122,10 @@ func (c *Config) Set(keys []string, value string) {
 // Read gh configuration files from the local file system and
 // return a Config.
 var Read = func() (*Config, error) {
-	var err error
 	once.Do(func() {
-		cfg, err = load(generalConfigFile(), hostsConfigFile())
+		cfg, loadErr = load(generalConfigFile(), hostsConfigFile())
 	})
-	return cfg, err
+	return cfg, loadErr
 }
 
 // ReadFromString takes a yaml string and returns a Config.

